torchreid/integration/sc/openvino_task.py

```diff
@@ -199,9 +170,40 @@ def evaluate(self,
                  evaluation_metric: Optional[str] = None):
         if evaluation_metric is not None:
             logger.warning(f'Requested to use {evaluation_metric} metric,'
-                            'but parameter is ignored. Use accuracy instead.')
+                           'but parameter is ignored. Use accuracy instead.')
         output_result_set.performance = MetricsHelper.compute_accuracy(output_result_set).get_performance()
 
+    def deploy(self,
+               output_path: str):
+        work_dir = os.path.dirname(demo.__file__)
+        model_file = inspect.getfile(type(self.inferencer.model))
```

Comment:
Seems like code in this method is mostly algo/model agnostic, so I guess it can be moved to a higher level entity.
Also I heard that Greg is working on some architecture proposals touching that stuff. May be it worth to reach him and discuss.

Reply:
I agree that code is mostly agnostic, but this depends on task environment as well as `export` and `optimize`. Export and optimize as much as deploy almost the same for all tasks. Interfaces for these methods contains only 1 abstract method. So at the moment OTE_SDK doesn't have any higher level entity. So now I am going create deploy for all tasks. If I  concoct new level I'll realize it. What about Greg, I add interface suggested by him in [PR](https://github.com/openvinotoolkit/training_extensions/pull/727) to ote